### PR TITLE
Some php doc revisions

### PR DESCRIPTION
**Pull request by Wilt**
Removed some unused class declarations 
Revised some php doc for parameters and dependencies
Called parent constructor in `HalJsonRenderer` class to set `$renderer`

### CODE REVIEW

File: src/Collection.php [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 use Traversable;
 use Zend\Stdlib\ArrayUtils;
+use Zend\Paginator\Paginator;
```
Comment (webimpress) [inline src/Collection.php]: alphabetical order please

File: src/Link/PaginationInjectorInterface.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 namespace ZF\Hal\Link;
 
 use ZF\Hal\Collection;
+use ZF\ApiProblem\ApiProblem;
```
Comment (webimpress) [inline src/Link/PaginationInjectorInterface.php]: alphabetical order please

File: src/Plugin/Hal.php [hunk 3]
```diff
@@ -39,6 +39,7 @@
 use ZF\Hal\Metadata\MetadataMap;
 use ZF\Hal\Resource;
 use ZF\Hal\ResourceFactory;
+use Traversable;
```
Comment (webimpress) [inline src/Plugin/Hal.php]: alphabetical order please

File: src/ResourceFactory.php [hunk 4]
```diff
@@ -7,13 +7,13 @@
...
 use Closure;
-use ZF\Hal\Collection;
-use ZF\Hal\Entity;
 use ZF\Hal\Extractor\EntityExtractor;
 use ZF\Hal\Exception;
 use ZF\Hal\Link\Link;
 use ZF\Hal\Link\LinkCollection;
 use ZF\Hal\Metadata\Metadata;
+use Traversable;
+use Zend\Paginator\Paginator;
```
Comment (webimpress) [inline src/ResourceFactory.php]: alphabetical order please

**Comment by Wilt** [comment]
@webimpress Done...

File: src/Module.php [hunk 5]
```diff
@@ -30,7 +31,9 @@ public function getConfig()
      */
     public function onBootstrap(MvcEvent $e)
     {
-        $events = $e->getTarget()->getEventManager();
+        /** @var Application $application */
```
Comment (weierophinney) [inline src/Module.php]: Should probably be the `ApplicationInterface` here.

File: src/ResourceFactory.php [hunk 6]
```diff
@@ -7,8 +7,8 @@
 namespace ZF\Hal;
 
 use Closure;
-use ZF\Hal\Collection;
-use ZF\Hal\Entity;
```
Comment (weierophinney) [inline src/ResourceFactory.php]: These are used in exactly the docblocks you're editing...

File: src/View/HalJsonStrategy.php [hunk 7]
```diff
@@ -32,7 +32,7 @@ class HalJsonStrategy extends JsonStrategy
      */
     public function __construct(HalJsonRenderer $renderer)
     {
-        $this->renderer = $renderer;
+        parent::__construct($renderer);
```
Comment (weierophinney) [inline src/View/HalJsonStrategy.php]: It's probably safer to leave this as it was. While the `HalJsonRenderer` *currently* extends the `JsonRenderer`, that's an implementation detail, and could change in the future.

File: src/ResourceFactory.php (hunk 6 above)
Comment (weierophinney) [inline src/ResourceFactory.php]: Of course, they're in the same namespace, so no change is necessary here. My bad!

**Comment by weierophinney** [comment]
Thanks, @Wilt

**Pull request merged**